contentctl/objects/content_versioning_service.py

```diff
@@ -0,0 +1,530 @@
+        ]
+
+    def _query_content_versioning_service(self, method: str, body: dict[str, Any] = {}) -> Record:
+        """
+        Queries the SA-ContentVersioning service. Output mode defaults to JSON.
+
+        :param method: HTTP request method (e.g. GET)
+        :type method: str
+        :param body: the payload/data/body of the request
+        :type body: dict[str, Any]
+
+        :returns: a splunklib Record object (wrapper around dict) indicating the response
+        :rtype: :class:`splunklib.data.Record`
+        """
+        # Add output mode to body
+        if "output_mode" not in body:
+            body["output_mode"] = "json"
+
+        # Query the content versioning service
+        try:
+            response = self.service.request(                                                        # type: ignore
+                method=method,
+                path_segment="configs/conf-feature_flags/general",
+                body=body,
+                app="SA-ContentVersioning"
+            )
+        except HTTPError as e:
+            # Raise on any HTTP errors
+            raise HTTPError(
+                f"Error querying content versioning service: {e}"
+            ) from e
+
+        return response
+
+    @property
+    def is_versioning_activated(self) -> bool:
+        """
+        Indicates whether the versioning service is activated or not
+
+        :returns: a bool indicating if content versioning is activated or not
+        :rtype: bool
+        """
+        # Query the SA-ContentVersioning service for versioning status
+        response = self._query_content_versioning_service(method="GET")
+
+        # Grab the response body and check for errors
+        if "body" not in response:
+            raise KeyError(
+                f"Cannot retrieve versioning status, 'body' was not found in JSON response: {response}"
+            )
+        body: Any = response["body"]                                                             # type: ignore
+        if not isinstance(body, ResponseReader):
+            raise ValueError(
+                "Cannot retrieve versioning status, value at 'body' in JSON response had an unexpected"
+                f" type: expected '{ResponseReader}', received '{type(body)}'"
+            )
+
+        # Read the JSON and parse it into a dictionary
+        json_ = body.readall()
+        try:
+            data = json.loads(json_)
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Unable to parse response body as JSON: {e}") from e
+
+        # Find the versioning_activated field and report any errors
+        try:
+            for entry in data["entry"]:
+                if entry["name"] == "general":
+                    return bool(int(entry["content"]["versioning_activated"]))
+        except KeyError as e:
+            raise KeyError(
+                "Cannot retrieve versioning status, unable to versioning status using the expected "
```

Comment:
Very minor typo here. I assume it is `Unable to determine versioning status` instead of `unable to versioning status`?

Reply:
Good catch!